src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,70 +1,243 @@
 {
-    "$defs": {
-        "dateTimePattern": {
-            "type": "string",
-            "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2})$"
-        }
+  "$defs": {
+    "dateTimePattern": {
+      "type": "string",
+      "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2})$"
+    },
+    "TimePattern": {
+      "type": "string",
+      "pattern": "^(\\d{2}:){3}\\d{2}$"
     },
-    "type": "object",
-    "properties": {
-        "workflow": {
-            "type": "object",
-            "properties": {
+    "BoolLoop": {
+      "type": "object",
+      "patternProperties": {
+        "^(and|or|not|nand|nor|xor)(_.*)?$": {
+          "$ref": "#/$defs/BoolLoop"
+        }
+      }
+    }
+  },
+  "type": "object",
+  "properties": {
+    "workflow": {
+      "type": "object",
+      "properties": {
+        "attrs": {
+          "type": "object",
+          "properties": {
+            "realtime": {
+              "type": "boolean"
+            },
+            "scheduler": {
+              "type": "string",
+              "enum": [
+                "sge",
+                "lsf",
+                "lsfcray",
+                "ll",
+                "torque",
+                "moabtorque",
+                "moab",
+                "pbspro",
+                "slurm",
+                "cobalt"
+              ]
+            },
+            "cyclethrottle": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "taskthrottle": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": [
+            "realtime",
+            "scheduler"
+          ]
+        },
+        "cycledefs": {
+          "type": "object",
+          "properties": {
+            "groupname": {
+              "type": "array",
+              "items": {
+                "$ref": "#/$defs/dateTimePattern"
+              },
+              "required": [
+                "groupname"
+              ]
+            },
+            "required": [
+              "groupname"
+            ]
```

Comment:
Have `groupname` required twice here. I think you only need it outside the `properties` section once